pkg/cmd/pulumi/config.go

```diff
@@ -1016,13 +1016,26 @@ func getStackConfiguration(
 	stack backend.Stack,
 	project *workspace.Project,
 	fallbackSecretsManager secrets.Manager, // optional
+) (backend.StackConfiguration, secrets.Manager, error) {
+	return getStackConfigurationWithFallback(ctx, stack, project, fallbackSecretsManager, nil)
+}
+
+func getStackConfigurationWithFallback(
+	ctx context.Context,
+	stack backend.Stack,
+	project *workspace.Project,
+	fallbackSecretsManager secrets.Manager, // optional
+	fallbackGetConfig func() (config.Map, error), // optional
 ) (backend.StackConfiguration, secrets.Manager, error) {
 	defaultStackConfig := backend.StackConfiguration{}
 
 	workspaceStack, err := loadProjectStack(project, stack)
 	if err != nil || workspaceStack == nil {
+		if fallbackGetConfig == nil {
+			return defaultStackConfig, nil, err
+		}
 		// On first run or the latest configuration is unavailable, fallback to check the project's configuration
-		cfg, err := backend.GetLatestConfiguration(ctx, stack)
+		cfg, err := fallbackGetConfig()
```

Comment:
If there are syntax errors in the yaml file, shouldn't we still be bubbling that error up to the user, even during destroy? I wonder if we should only be doing this fallback if the stack config doesn't exist.

Reply:
I've updated the PR so that `pulumi destroy` will only fall back on the latest config if both:
1. `--stack` is provided
2. there's no project detected (which occurs when outside of a pulumi project directory)